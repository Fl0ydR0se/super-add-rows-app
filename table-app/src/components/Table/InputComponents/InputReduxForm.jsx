import React from "react";
import { Field } from "redux-form";
import {form, table, button} from "../Table.module.css";
import { Button } from "@material-ui/core";
import {maxLength, required} from '../../Validation/validation-form'
import {InputTextArea} from './InputTextArea'

const maxLength15 = maxLength(15);

export const InputForm = props => {
  let {
    disableInput,
    handleSubmit,
    invalid,
    pristine,
    reset,
    submitting
  } = props;

  return (
      <form className={form} onSubmit={handleSubmit}>
        <div className={table}>
          <Field
            validate={[maxLength15, required]}
            name="row"
            component={InputTextArea}
            label="Add new row"
            disabled={disableInput}
          />
        </div>
        <div className={button}>
          <Button
            color="primary"
            type="submit"
            disabled={pristine || submitting || invalid}
          >
            {"Add row"}
          </Button>
          <Button
            color="secondary"
            type="button"
            disabled={pristine || submitting}
            onClick={reset}
          >
            {"Clear Input"}
          </Button>
        </div>
      </form>
  );
};

