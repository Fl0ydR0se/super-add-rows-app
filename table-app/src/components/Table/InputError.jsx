import React from "react";
import {beautyText} from './Table.module.css'

export const InputError = ({ error }) => {
  return <div className={beautyText}>{error}</div>;
};
