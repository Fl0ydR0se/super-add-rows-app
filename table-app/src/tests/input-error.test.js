import React from "react";
import { InputError } from "../components/Table/InputComponents/InputError";
import { configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

describe("Input tests", () => {
  test("Input error ", () => {
    const propsErrorText = "error text";
    const component = mount(<InputError error={propsErrorText}></InputError>);
    expect(component.props().error).toEqual(propsErrorText);
    expect(component.find("div")).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });
});
