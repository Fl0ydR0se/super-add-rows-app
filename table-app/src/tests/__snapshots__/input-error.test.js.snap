// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input tests Input error  1`] = `ReactWrapper {}`;
